text/0000-array-trait.md

```diff
@@ -0,0 +1,152 @@
+- Feature Name: array-trait
+- Start Date: 2017-02-21
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Extend and stabilize the `FixedSizeArray` trait,
+as a stop-gap solution for integer parameters in generics.
+
+
+# Motivation
+[motivation]: #motivation
+
+One of Rust’s basic families of types is fixed-size arrays:
+`[T; N]` where `T` is any (statically-sized) type, and `N` is an (`usize`) integer.
+Arrays with the same `T` item type but of different sizes are themselves different types.
+
+A [long-standing](https://github.com/rust-lang/rfcs/pull/884)
+feature [request](https://github.com/rust-lang/rfcs/issues/1038)
+it the ability to write code that is generic over the size of an array.
+The solution typically proposed for this it “type-level integers”,
+where generic items can have integers as a new *kind* of type-level parameters,
+in addition to lifetime parameters and type parameters.
+
+This by itself introduce it’s share of complexity.
+Even so, such proposals are often extended to cover arbitrary type levels (not just integers),
+or even dependent types.
+As a result, every RFC on the subject so far has been postponed:
+[1](https://github.com/rust-lang/rfcs/pull/884),
+[2](https://github.com/rust-lang/rfcs/pull/1062),
+[3](https://github.com/rust-lang/rfcs/pull/1520),
+[4](https://github.com/rust-lang/rfcs/issues/1557),
+[5](https://github.com/rust-lang/rfcs/pull/1657).
+
+This RFC propose an addition to the language that,
+although less theoretically pleasing,
+is much smaller and therefore hopefully more likely to be accepted.
+Additionally, it does not prevent a more general solution from also happening in the future.
+
+Arrays are the only case where integers occur in types, in current Rust.
+Therefore a solution specifically for arrays
```

Comment:
On re-reading you comment it sounds like it doesn’t involve arrays, but it might involve type-level integers? Yes, this RFC does not entirely remove the need for type-level integers, but it doesn’t prevent adding them later either.